drud/ddev
# Fix database corruption problem on ddev rm

## The Problem/Issue/Bug:

In the recent Drupal-2-geddon updates I noted that a few of my sites (perhaps with larger dbs) would fail to start after ddev start, ddev import-db, ddev rm, ddev start. After further investigation I found that there was database corruption as described in https://github.com/drud/ddev/issues/748 (which is about the situation of a reboot or docker exiting). 

## How this PR Solves The Problem:

* We weren't stopping containers before forcing their removal, so db container didn't get a chance to clean up and exit correctly. I'm sure we used to, but in our effort to clean up when docker-compose.yaml is missing, we apparently removed that. 
* Added to docker-compose.yaml an increased exit timeout for the db container (default is 10s, that normally is totally adequate, but it doesn't do any harm to have it higher).

## Manual Testing Instructions:

With a larger site (larger db) do `ddev start`, `ddev import-db`, `ddev rm`, `ddev start`. It should work without trouble.

## Automated Testing Overview:


I can write a test for this, but it seems quite heavyweight. Not sure.

## Related Issue Link(s):

Definitely relates to #748, which is about corruption when docker goes away (like at reboot of host).

https://github.com/drud/mariadb-local/pull/52 was a PR aimed at fixing #748 (it failed, but was interesting). 

## Release/Deployment notes:
